index.rst

```diff
@@ -4,9 +4,149 @@ File Formats and Layouts for Cell-based Coadds
+
+The Green Bank convention makes no mention of having multiple images as different columns in the same table, however, so we may want to consider putting different image planes in different binary table HDUs, each with corresponding rows, and probably some columns - like the WCS ones - duplicated.
+The extra storage cost of another header and some duplicated columns is insignificant, and it may be desirable to put multiple cells from the same plane closer together on disk than different planes of the same cell, especially since we expect that subimage reads of some planes (i.e. the image data image) will be much more popular than others (e.g. noise realizations).
+
+PSF images fit naturally in this layout, as there's nothing wrong with having some array columns having a different shape.
+There might be some confusion with a Green Bank convention WCS, though, if we go with a single table with different columns for different image planes as well as the PSF; there's no way to mark which image columns the WCS applies to.
+
+The main problem with binary table layouts is compression support.
+The latest FITS standard does include (in section 10.3) a discussion of "tiled table compression", which seems like it'd be sufficient, as long as compressing one cell at a time is enough to get a good compression ratio (this is unclear).
+Unlike image tile compression, binary table tile compression doesn't support lossy compression algorithms or dithered quantization, but it would still be possible to do our own non-dithered quantization and use ``BZERO`` and ``BSCALE`` to record the transformation from integer back to floating-point.
+The bigger problem is that there does not appear to be any implementations of it: there is no mention of it in either the CFITSIO or Astropy documentation (and even if an implementation does exist in, say, the Java ecosystem, we wouldn't be in a position to use it).
+While we've already discussed the fact that we'll probably need to implement some low-level FITS image tile compression code in order to do decompressed subimage reads with object stores anymore, the binary table compression situation is much more problematic:
+
+- tables are much more complicated than images;
+- we would have to implement writes ourselves, not just reads;
+- we would not have a reference implementation we could use for testing;
+- if the standard has not seen real use, we stand a good chance of discovering uncovered edge cases or other defects;
+- third-party FITS readers would definitely not be able to read our files, at least not without significant work.
+
+In fact, even without compression, the binary table layout would require writing our code just to solve the problem of subimage reads over object stores, since Astropy cannot do efficient table subset reads and CFITSIO cannot do object store reads.
+
+Per-HDU Cells
```

Comment:
There will be a lot of cells and I've seen serious performance issues in FITS files with many many HDUs. I'd be wary of this one.

Reply:
My mental model is that this is because you have to seek through the whole file to find the byte addresses where each HDU stars, and I think an address table will fix this _for readers that use it_.  But it's still a problem for other readers that are unaware of it.

Reply:
That might have been a python `fitsio` issue only, not an issue with `cfitsio`.   I think the problem was fixed a while back, but if you think this persists please make an issue.